packages/webmonetization-polyfill-utils/src/lib/MonetizationTagObserver.ts

```diff
@@ -158,6 +164,15 @@ export class MonetizationTagObserver {
     }
 
     const details = this.getPaymentDetails(meta)
+
+    if (this.metaTags.size + 1 > this.maxMetas) {
```

Comment:
I believe at one point in the design phase the code was intended to handle the case of tag 2 starting once tag 1 was removed, which would have been a overwrought, but that may explain why we were tracking the tags in the Map BEFORE throwing errors